WordPress/gutenberg
# Block Library: Add core template part block.

Follows #18339

## Description

This PR adds the implementation for the `core/template-part` block introduced in #18339.

It focuses on the server-side render callback so that block templates that use template parts begin to render correctly on the front end. The editor implementation for creating and editing template parts will come in a follow-up PR.

## How has this been tested?

It was verified that the following steps make post previews render as expected:

<img width="565" alt="Screen Shot 2019-11-25 at 4 11 13 PM" src="https://user-images.githubusercontent.com/19157096/69588701-51cf2200-0f9e-11ea-9de8-deb6c705fe45.png">

- Enable the Gutenberg full site editing experiment.
- Add the following files to your active theme's folder:

`/block-templates/single.html`:
```html


<p>Single Template</p>

```

`/block-template-parts/header.html`:
```html

<p>Header TwentyTwenty Template Part</p>

```

## Types of Changes

*New Feature:* Added a new server-rendered template part block for the full site editing experiment.

## Checklist:
